hicexplorer/hicFindTADs.py

```diff
@@ -1190,10 +1201,24 @@ def min_pvalue(self, min_idx):
                     pval = np.nan
             pvalues += [pval]
 
+        # print("pvalues 1185", pvalues)
+
         assert len(pvalues) == len(new_min_idx)
-        # bonferroni correction
-        pvalues = np.array(pvalues) * len(pvalues)
-        pvalues[np.array([e > 1 if ~np.isnan(e) else False for e in pvalues])] = 1
+
+        # fdr
+        if self.multipleComparisons == 'fdr':
```

Comment:
did you compare the resulting fdr values with the result from R just to be sure that there are no problems?

Reply:
I did not so far and will do it in the next days. I used https://matthew-brett.github.io/teaching/fdr.html as a reference.

Reply:
I am close, but not close enough :( I will debug it and get back to you.

|q-value|R fdr | my fdr|
|----------|-------|---------|
|0.01     |256  |253     |               
|0.02     |335  |329     |
|0.03     |388  |367     |
|0.04     |413  |403     |
|0.05     |425  |414     |
|0.1       |489  |470     |
|0.2       |537  |515     |

Reply:
If only the fdr is considered the numbers are equal. The difference and the more restrictive behaviour comes from the check against the membership of `delta_of_min` [Line 911](https://github.com/joachimwolff/HiCExplorer/blob/468608a2d4372e7cf97afd6f33f1014538779be5/hicexplorer/hicFindTADs.py#L911) and the subsequent test `if delta_of_min[idx] >= self.delta` [Line 914](https://github.com/joachimwolff/HiCExplorer/blob/468608a2d4372e7cf97afd6f33f1014538779be5/hicexplorer/hicFindTADs.py#L914).
I think this test should be included for the fdr too. Nevertheless, I discovered a few potential bugs in this and changed it in PR #122